Fix property handling with SAJ

jvalkeal opened the pull request:
- Now as a quick workaround to get metrics working,
  conditionally add normal properties reflecting i.e.
  SPRING_CLOUD_APPLICATION_GUID as spring.cloud.application.guid.
- See from added comments in code why it wasn't possible to
  remove those env style formats.
- This change now allows scdf server to show metrics again.
- Fixes #109

jvalkeal committed:
Fix property handling with SAJ

- Now as a quick workaround to get metrics working,
  conditionally add normal properties reflecting i.e.
  SPRING_CLOUD_APPLICATION_GUID as spring.cloud.application.guid.
- See from added comments in code why it wasn't possible to
  remove those env style formats.
- This change now allows scdf server to show metrics again.
- Fixes #109

mminella commented on spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalAppDeployer.java:
```diff
@@ -175,6 +175,17 @@ public String deploy(AppDeploymentRequest request) {
+					appInstanceEnv.put("spring.application.index", Integer.toString(i));
+					appInstanceEnv.put("spring.cloud.application.guid", Integer.toString(port));
+				}
+
```
Shouldn't the other way of setting properties be in an else here?

jvalkeal replied:
I put a comment to PR to read reason for that from comment in code as I was 100% sure someone would ask it. In nutshell, tests break if we take it away.

mminella replied:
So we should fix the test to be valid under the new conditions.  And since you bring it up, we should have a test for this scenario as well.

jvalkeal replied:
Maybe that `spi-test-app` could be enhanced to track these. We're actually missing integration tests for metric collector as integration with that was only thing what got broken what this change is fixing.

mminella replied:
Why can't we validate these in a similar way to the way we do with the command builders like here: https://github.com/spring-cloud/spring-cloud-deployer-local/blob/master/spring-cloud-deployer-local/src/test/java/org/springframework/cloud/deployer/spi/local/JavaExecutionCommandBuilderTests.java#L168

jvalkeal replied:
That sounds reasonable. It doesn't test the actual problem for spring resolving cross refs, afaik, but is better than nothing.

markpollack replied:
I agree, should be an `else` cause as it just looks confusing if anyone starts to debug to see the same property set twice (once in env-var and once in SAJ).

jvalkeal committed:
Address review comments

- Use deployer as 1.3.2.BUILD-SNAPSHOT fix for
  INSTANCE_INDEX vs. instance.index needed for tests
  are are in its test app.
- Duplicate LocalAppDeployerIntegrationTests as
  LocalAppDeployerEnvironmentIntegrationTests order to define
  deployer props not to use SAJ. These tests now
  check INSTANCE_INDEX vs. instance.index.

mminella commented on spring-cloud-deployer-local/src/test/java/org/springframework/cloud/deployer/spi/local/LocalAppDeployerEnvironmentIntegrationTests.java:
```diff
@@ -0,0 +1,258 @@
+			// tweak random dir name on win to be shorter
+			String uuid = UUID.randomUUID().toString();
+			long l = ByteBuffer.wrap(uuid.toString().getBytes()).getLong();
+			return name.getMethodName() + Long.toString(l, Character.MAX_RADIX);
```
Should we move this up to the parent?  I can do this on merge if you're ok with it.
